wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginMetaRetriever.java

```diff
@@ -87,11 +93,23 @@ public PluginMetaRetriever(@Nullable @Named(PLUGIN_REGISTRY_PROPERTY) String plu
    */
   @Beta
   public Collection<PluginMeta> get(Map<String, String> attributes) throws InfrastructureException {
-    if (pluginRegistry == null || attributes == null || attributes.isEmpty()) {
+    // Have to check for empty value instead of plain null as it's possible to have empty
+    // plugins attribute
+    String pluginsAttribute =
```

Comment:
Could you also check that map is not null? Otherwise, we would have an NPE I suppose.

Reply:
Great catch, thanks!